### PR TITLE
feat: add new OG image for contributor highlights

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds new edge function for generating an OpenGraph image for contributor highlights

## Related Tickets & Documents

Closes https://github.com/open-sauced/app/issues/3806

## Mobile & Desktop Screenshots/Recordings

image

## Steps to QA

1. Open Deploy Preview
2. Go to `/feed` page
3. Pick a highlight, and get the ID
4. Visit https://deploy-preview-3779--oss-insights.netlify.app/og-images/highlight/298 or with another highlight ID

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

**Review by nickytonline — COMMENTED** [review]
I haven't seen the design, but the heart should be aligned with the content and need a red or orange like our reaction colour.

CleanShot 2024-07-25 at 16 58 31

File: pages/feed/index.tsx [hunk 1]
```diff
@@ -62,7 +62,10 @@ export default function Feeds(props: HighlightSSRProps) {
   const topRef = useRef<HTMLDivElement>(null);
   const singleHighlight = props.highlight;
   const ogImage = props?.highlight
-    ? `${process.env.NEXT_PUBLIC_OPENGRAPH_URL}/highlights/${props.highlight.id}`
+    ? `${new URL(
+        `/og-images/highlight/${props.highlight.id}`,
+        process.env.NEXT_PUBLIC_BASE_URL || "http://localhost:3000"
+      )}`
```
Comment (jpmcb) [inline pages/feed/index.tsx:68]: Testing locally, I get a 404 on: http://localhost:3000/og-images/highlight/301

Is this the right endpoint to hit?

Comment (nickytonline) [reply pages/feed/index.tsx:68]: You need to run `netlify dev` ([Netlify CLI](https://docs.netlify.com/cli/get-started/)) if testing OG images locally.

Comment (jpmcb) [reply pages/feed/index.tsx:68]: 🫠 gotcha! Thanks Nick!

Comment (brandonroberts) [reply pages/feed/index.tsx:68]: Here's a deploy preview URL: https://deploy-preview-3779--oss-insights.netlify.app/og-images/highlight/298

**Review by jpmcb — APPROVED** [review]
Screenshot 2024-07-30 at 9 26 55 AM

**Review by nickytonline — APPROVED** [review]
Looks great @brandonroberts! 🚢

**Pull request merged**